### DISCUSSION TITLE
1.11.11 P-chain fee config

### DISCUSSION THREAD

**Discussion by mikkompeltonen** [ava-labs/avalanchego#3378] [Q&A]
The release of 1.11.11 introduced getFeeState and getFeeConfig APIs for the P-chain. Are these in preparation for the dynamic fees upgrade? Can we use these to determine all gas fees for P-chain operations already? If so, where can we find documentation on them?

**Comment by StephenButtolph** [comment]
`platform.getFeeState` and `platform.getFeeConfig` APIs are only useful on networks with the dynamic fees upgrade activated. (Such as local test networks).

**Comment by meaghanfitzgerald** [comment]
> Are these in preparation for the dynamic fees upgrade?

Yes

> Can we use these to determine all gas fees for P-chain operations already?

_Insert stephen's answer here_

If you [install avalanche-cli](https://github.com/ava-labs/avalanchego/discussions/3378#discussioncomment-10604334) and run `avalanche network start` the cli will automatically run a local network with the latest release. You can search through additional CLI commands in [this tool](https://l1cli.com/) built by community member @m5k. 

> If so, where can we find documentation on them?

GetFeeConfig is tested [here](https://github.com/ava-labs/avalanchego/blob/8974ed8700b367c8b6bbbc325e5e14d7a606e482/vms/platformvm/service_test.go#L1124) and implemented by the wallet [here](https://github.com/ava-labs/avalanchego/blob/bee532d997f6a3c0bf2fe7bc4271abd81bf93feb/wallet/chain/p/context.go#L45).

**Reply by meaghanfitzgerald** [reply]
Once the network is running locally you can hit the endpoint via curl or postman: 
```
curl -X POST --data '{
    "jsonrpc": "2.0",
    "method": "platform.getFeeState",
    "params": {},
    "id": 1
}' -H 'content-type:application/json;' 127.0.0.1:9650/ext/bc/P
```
example response: 
```
{"jsonrpc":"2.0","result":{"capacity":84270,"excess":466,"price":1,"timestamp":"2024-09-05T08:17:19-03:00"},"id":1}
```

**Reply by meaghanfitzgerald** [reply]
All P-chain APIs activated on Mainnet are documented [here](https://docs.avax.network/api-reference/p-chain/api)
